### PR TITLE
chore: Pre-release version 2.4.0-beta.2

### PR DESCRIPTION
**Pull request by KaylaNguyen**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -121,8 +121,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Added
 - Initial release
 
-[Unreleased]: https://github.com/GoogleCloudPlatform/functions-framework-python/compare/v2.4.0-beta.1...HEAD
```
Comment (grant) [inline CHANGELOG.md:124]: I believe this should be: https://github.com/GoogleCloudPlatform/functions-framework-python/compare/v2.4.0-beta.2...HEAD

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,7 +6,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
-## [2.4.0-beta.1] - 2021-10-29
```
Comment (grant) [inline CHANGELOG.md:9]: I don't think we should delete previous changes, even if pre-release.

**Review by grant — CHANGES_REQUESTED** [review]
- I'm unsure if we want to delete the beta.1 release
- nit: I believe these commits should be in the type `chore`, as in `chore: Pre-release version 2.4.0-beta.2` as to not produce a change in semver.

File: CHANGELOG.md (hunk 2 above)
Comment (KaylaNguyen) [reply CHANGELOG.md:9]: Done.

**Review by grant — APPROVED** [review]
LGTM 🥳

**Pull request merged**